### PR TITLE
added better IT tests for binary content

### PR DESCRIPTION
**Pull request by acoburn**
This partially addresses https://jira.duraspace.org/browse/FCREPO-1269

This commit added better IT tests for binaries and containers

It also adds much imporved integration tests with an external triplestore and an external Solr instance

renamed Namespaces class so as not to conflict with the o.a.c.builder.xml.Namespaces class

What remains for the JIRA issue is some OSGi testing.

### CODE REVIEW

File: src/main/java/org/fcrepo/camel/processor/SparqlDeleteProcessor.java [hunk 1]
```diff
@@ -40,7 +40,8 @@ public void process(final Exchange exchange) throws IOException {
         final Message in = exchange.getIn();
         final String subject = ProcessorUtils.getSubjectUri(in);
 
-        in.setBody("DELETE WHERE { <" + subject + "> ?p ?o }");
+        in.setBody("DELETE WHERE { <" + subject + "> ?p ?o };\n" +
```
Comment (awoods) [inline src/main/java/org/fcrepo/camel/processor/SparqlDeleteProcessor.java]: This looks like a (valid) hack for plucking out specific triples. If so, please add a comment to the effect of why you are requiring the fcr:export path element. There is a good chance that the jcr/xml export may be going away at some point, and it is important that we know if this line of code is a true dependency.

File: src/main/java/org/fcrepo/camel/processor/SparqlUpdateProcessor.java [hunk 2]
```diff
@@ -58,6 +58,7 @@ public void process(final Exchange exchange) throws IOException {
         serializer.serialize(serializedGraph, graph.getGraph(), "text/rdf+nt");
 
         exchange.getIn().setBody("DELETE WHERE { <" + subject + "> ?p ?o };\n" +
+                                 "DELETE WHERE { <" + subject + "/fcr:export?format=jcr/xml> ?p ?o };\n" +
```
Comment (awoods) [inline src/main/java/org/fcrepo/camel/processor/SparqlUpdateProcessor.java]: This looks like a (valid) hack for plucking out specific triples. If so, please add a comment to the effect of why you are requiring the fcr:export path element. There is a good chance that the jcr/xml export may be going away at some point, and it is important that we know if this line of code is a true dependency.

File: pom.xml [hunk 3]
```diff
@@ -557,6 +649,10 @@
                   <name>integration-test</name>
                   <value>true</value>
                 </property>
+                <!--<property>-->
```
Comment (awoods) [inline pom.xml]: Removed commented out block.

File: pom.xml [hunk 4]
```diff
@@ -397,10 +399,97 @@
       </plugins>
     </pluginManagement>
 
-
     <defaultGoal>install</defaultGoal>
 
     <plugins>
+      <!-- Launch jetty for unit testing with Solr -->
```
Comment (awoods) [inline pom.xml]: Should be:

Comment (acoburn) [inline pom.xml]: @awoods was there a comment here?

File: src/main/java/org/fcrepo/camel/processor/SparqlDeleteProcessor.java (hunk 1 above)
Comment (acoburn) [inline src/main/java/org/fcrepo/camel/processor/SparqlDeleteProcessor.java]: @awoods I'll add a note to this effect.

File: pom.xml (hunk 4 above)
Comment (awoods) [inline pom.xml]: Markdown hid the previous comment.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4-labs/fcrepo-camel/commit/a361ca163f4ee6a910f1fde9f72f879f0d4e123f

**Pull request closed without merging**